go/mysql/binlog_event_mysql56_test.go

```diff
@@ -94,46 +99,108 @@ func TestMysql56GTID(t *testing.T) {
 	tableMap := &TableMap{}
-	require.True(t, mysql56TransactionPayloadEvent.IsTransactionPayload())
-
-	// The generated event is the result of the following SQL being executed in vtgate
-	// against the commerce keyspace:
-	// begin; insert into customer values (1, "mlord@planetscale.com"), (2, "sup@planetscale.com"); commit;
-	// All of these below internal events are encoded in the compressed transaction
-	// payload event.
-	want := []string{
-		"BEGIN",                     // Query event
-		"vt_commerce.customer",      // TableMap event
-		"[1 mlord@planetscale.com]", // WriteRows event
-		"[2 sup@planetscale.com]",   // WriteRows event
-		"COMMIT",                    // XID event
+	tmpDir := os.TempDir()
+	tmpFilePrefix := tmpFilePattern[0 : len(tmpFilePattern)-1] // Without the trailing *
+
+	testCases := []struct {
+		name     string
+		event    BinlogEvent
+		want     []string
+		inMemory bool
+	}{
+		{
+			name:  "Small event in memory",
+			event: NewMysql56BinlogEvent([]byte{0xc7, 0xe1, 0x4b, 0x64, 0x28, 0x5b, 0xd2, 0xc7, 0x19, 0xdb, 0x00, 0x00, 0x00, 0x3a, 0x50, 0x00, 0x00, 0x00, 0x00, 0x02, 0x01, 0x00, 0x03, 0x03, 0xfc, 0xfe, 0x00, 0x01, 0x01, 0xb8, 0x00, 0x28, 0xb5, 0x2f, 0xfd, 0x00, 0x58, 0x64, 0x05, 0x00, 0xf2, 0x49, 0x23, 0x2a, 0xa0, 0x27, 0x69, 0x0c, 0xff, 0xe8, 0x06, 0xeb, 0xfe, 0xc3, 0xab, 0x8a, 0x7b, 0xc0, 0x36, 0x42, 0x5c, 0x6f, 0x1b, 0x2f, 0xfb, 0x6e, 0xc4, 0x9a, 0xe6, 0x6e, 0x6b, 0xda, 0x08, 0xf1, 0x37, 0x7e, 0xff, 0xb8, 0x6c, 0xbc, 0x27, 0x3c, 0xb7, 0x4f, 0xee, 0x14, 0xff, 0xaf, 0x09, 0x06, 0x69, 0xe3, 0x12, 0x68, 0x4a, 0x6e, 0xc3, 0xe1, 0x28, 0xaf, 0x3f, 0xc8, 0x14, 0x1c, 0xc3, 0x60, 0xce, 0xe3, 0x1e, 0x18, 0x4c, 0x63, 0xa1, 0x35, 0x90, 0x79, 0x04, 0xe8, 0xa9, 0xeb, 0x4a, 0x1b, 0xd7, 0x41, 0x53, 0x72, 0x17, 0xa4, 0x23, 0xa4, 0x47, 0x68, 0x00, 0xa2, 0x37, 0xee, 0xc1, 0xc7, 0x71, 0x30, 0x24, 0x19, 0xfd, 0x78, 0x49, 0x1b, 0x97, 0xd2, 0x94, 0xdc, 0x85, 0xa2, 0x21, 0xc1, 0xb0, 0x63, 0x8d, 0x7b, 0x0f, 0x32, 0x87, 0x07, 0xe2, 0x39, 0xf0, 0x7c, 0x3e, 0x01, 0xfe, 0x13, 0x8f, 0x11, 0xd0, 0x05, 0x9f, 0xbc, 0x18, 0x59, 0x91, 0x36, 0x2e, 0x6d, 0x4a, 0x6e, 0x0b, 0x00, 0x5e, 0x28, 0x10, 0xc0, 0x02, 0x50, 0x77, 0xe0, 0x64, 0x30, 0x02, 0x9e, 0x09, 0x54, 0xec, 0x80, 0x6d, 0x07, 0xa4, 0xc1, 0x7d, 0x60, 0xe4, 0x01, 0x78, 0x01, 0x01, 0x00, 0x00}),
+			// The generated event is the result of the following SQL being executed in vtgate
+			// against the commerce keyspace:
+			// begin; insert into customer values (1, "mlord@planetscale.com"), (2, "sup@planetscale.com"); commit;
+			// All of these below internal events are encoded in the compressed transaction
+			// payload event.
+			want: []string{
+				"BEGIN",                     // Query event
+				"vt_commerce.customer",      // TableMap event
+				"[1 mlord@planetscale.com]", // WriteRows event
+				"[2 sup@planetscale.com]",   // WriteRows event
+				"COMMIT",                    // XID event
+			},
+			inMemory: true,
+		},
+		{
+			name:  "Large event using a temporary file",
```

Comment:
How feasible is it to add a `vplayer` test to simulate events which will use the temporary file logic? That will let us test both vstreamer and vplayer paths and with multiple such events.

We would need to make `zstdInMemoryDecompressorMaxSize` public so that we can use smaller values and enhance the blob unit tests, for example, to exercise both the in-memory and file based logic.

Reply:
vplayer gets the vevents after any decompression is done in vstreamer. We don't create or stream any compressed transaction payload events (only the events within them).

Reply:
We do create a `vstreamer` in the unit tests and parse the binlog events generated by the dmls in the unit test. If you add logs to the vstreamer code path, they can be seen in the test output ... 
```
streamerEngine = vstreamer.NewEngine(env.TabletEnv, env.SrvTopo, env.SchemaEngine, nil, env.Cells[0])
streamerEngine.InitDBConfig(env.KeyspaceName, env.ShardName)
streamerEngine.Open()
```

However we don't seem to be testing with compressed binlogs anywhere, so not required here.

Reply:
Yes, we're not using a vstreamer in these unit tests. We're testing the full path in the end-to-end tests for all of the event types.